Search/Header.swift

```diff
@@ -37,34 +37,51 @@ public class Header: NSObject {
         self.query = query
     }
 
+    //MARK: JSON initialiser
     convenience init?(json: JSON) {
-        guard let uri = json[Header.UriKey].string,
-            lr = json[Header.LrKey].string,
-            dataset = json[Header.DatasetKey].string,
-            outputSrs = json[Header.OutputSrsKey].string,
-            epoch = json[Header.EpochKey].string,
-            format = json[Header.FormatKey].string,
-            query = json[Header.QueryKey].string,
-            matchPrecision =  json[Header.MatchprecisionKey].number,
-            maxresults = json[Header.MaxresultsKey].number,
-            totalresults = json[Header.TotalresultsKey].number,
-            offset = json[Header.OffsetKey].number
-            else {
-                return nil
+        guard let
+            lr = json.stringValueForKey(Header.LrKey),
+            dataset = json.stringValueForKey(Header.DatasetKey),        
+            uri = json.stringValueForKey(Header.UriKey),
+            outputSrs = json.stringValueForKey(Header.OutputSrsKey),        
+            epoch = json.stringValueForKey(Header.EpochKey),
+            format = json.stringValueForKey(Header.FormatKey),        
+            query = json.stringValueForKey(Header.QueryKey)
+        else {
+            return nil
         }
+        let matchprecision = json.intValueForKey(Header.MatchprecisionKey)
+        let maxresults = json.intValueForKey(Header.MaxresultsKey)
+        let totalresults = json.intValueForKey(Header.TotalresultsKey)
+        let offset = json.intValueForKey(Header.OffsetKey)
 
         self.init(
-            matchprecision: Int(matchPrecision),
+            matchprecision: Int(matchprecision),            
```

Comment:
Also, probably don't need this any more do we? Might be able to remove this in the template. And get rid of the trailing white space. There is a setting in Xcode to get it to trim that for you though.